app/volume.go

```diff
@@ -311,10 +311,26 @@ func (job *Job) backupAndCleanup() (err error) {
 	}
 
 	// CronJob template has covered the credential already, so we don't need to get the credential secret.
-	if _, err := job.engine.SnapshotBackup(job.snapshotName, job.backupTarget, job.labels, nil); err != nil {
+	backupName, err := job.engine.SnapshotBackup(job.snapshotName, job.backupTarget, job.labels, nil)
+	if err != nil {
 		return err
 	}
+
 	target := engineapi.NewBackupTarget(job.backupTarget, job.engineImage, nil)
+
+	// Wait for backup creation complete
+	for {
```

Comment:
Do we need to add max wait count here?

Reply:
We should query the engine status instead of backup store since the backup may never show up if it errors out, but we cannot know how long to wait since we don't know how long it would take for very big backups.